src/views/MoviesPage.vue

```diff
@@ -0,0 +1,130 @@
+<script>
+import { defineComponent } from 'vue';
+
+import BaseTextInput from '@/components/global/BaseTextInput.vue';
+import BaseSelect from '@/components/global/BaseSelect.vue';
+import MoviesContainer from '@/components/MoviesPage/MoviesContainer.vue';
+
+import { useMoviesStore } from '@/store/movies';
+const moviesStore = useMoviesStore();
+
+export default defineComponent({
+  name: 'MoviesPage',
+  components: {
+    BaseTextInput,
+    BaseSelect,
+    MoviesContainer
+  },
+  data() {
+    return {
+      searchQuery: '',
+      movieCategory: ''
+    };
+  },
+  computed: {
+    movies() {
+      return moviesStore.allMovies;
+    },
+    genres() {
+      return [...new Set(moviesStore.genres)];
+    },
+    moviesByGenre() {
+      return this.movieCategory === '' ||
+        this.movieCategory === 'All Categories'
+        ? this.movies
+        : this.movies.filter(
+            (movie) => movie.genre.name === this.movieCategory
+          );
+    },
+    filteredMovies() {
```

Comment:
I like the splitting of responsibility in the `moviesByGenre` and `filteredMovies` functions 🏅